transport/src/main/java/io/netty/channel/ManualIoEventLoop.java

```diff
@@ -651,11 +663,16 @@ public long delayNanos(long currentTimeNanos) {
         public long deadlineNanos() {
             assert inEventLoop();
             long next = nextScheduledTaskDeadlineNanos();
-            long maxDeadlineNanos = ticker.nanoTime() + maxBlockingNanos;
-            if (next == -1) {
-                return maxDeadlineNanos;
+            if (maxBlockingNanos == Long.MAX_VALUE) {
+                // next == -1? -1 : next i.e. return next
+                return next;
+            }
+            long now = ticker.nanoTime();
+            // we cannot just check Math.min as nanoTime can be negative or wrap around!
```

Comment:
As explained at https://github.com/netty/netty/pull/15937/files#r2567235475

Reply:
@franz1981 that said saving `Math.min` is kind of a drop in the ocean ...